torch/_logging/_internal.py

```diff
@@ -5,17 +5,33 @@
 import re
 from dataclasses import dataclass, field
 from importlib import __import__
-from typing import Dict, Set
+from typing import Dict, Set, Callable, List
 from weakref import WeakSet
 
 log = logging.getLogger(__name__)
 
 DEFAULT_LOG_LEVEL = logging.WARN
-DEFAULT_FORMATTER = logging.Formatter(
-    "[%(asctime)s] %(name)s: [%(levelname)s] %(message)s"
-)
 LOG_ENV_VAR = "TORCH_LOGS"
 
+FORMATTER_PREFIXES : List[Callable[[], str]] = []
+
+class ExtendableLogFormatter(logging.Formatter):
+    def __init__(self, fmt=None, datefmt=None):
+        super().__init__(fmt, datefmt)
+
+    def format(self, record):
+        if len(FORMATTER_PREFIXES) == 0:
+            return super().format(record)
+        else:
+            prefix = ""
+            for fn in FORMATTER_PREFIXES:
+                prefix += fn()
+        record.msg = prefix + record.msg
+        return super().format(record)
+
+DEFAULT_FORMATTER = ExtendableLogFormatter(
```

Comment:
Since we are planning on onboarding other components to this, we should make the formatter an argument in _registrations.py, and set this for the pt2 loggers, rather than the default for everyone. I don't think every component in pytorch will want this.